### ISSUE TITLE
SyntaxError: Cannot use import statement outside a module

### ISSUE THREAD

**Issue by markocir** [trailheadapps/ebikes-lwc#465]
### Summary

I was trying to runt tests `npm run test:ui` and it failed with: `SyntaxError: Cannot use import statement outside a module`.
I have changed ChromeDriver version to latest and run npm install and it did not change the outcome:` "chromedriver": "^107.0.3",`

### Salesforce Org Type

Scratch Org

### Steps To Reproduce

1. Installing E-Bikes using a Scratch Org
3. `npm install`
4. `npm run test:ui:compile`
5. `npm run test:ui:generate:login`
6. `npm run test:ui` **-> this fails**
7. update `chromedriver` dependency version to `^107.0.3`
8. `npm install`
9. `npm run test:ui` **-> this still fails**

### Relevant Log Output

![image](https://user-images.githubusercontent.com/18015314/200170721-96b52c59-32c1-4144-b148-7a73dd0972ab.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pozil** [comment]
Hi @markocir. Thanks for reporting this issue but I couldn't reproduce it.
Can you confirm that you're using node v16 or above?
I just updated the dependencies on the repo, just in case.

**Comment by markocir** [comment]
Hi,
I have node version v16.15.0. Also I am using wsl2 subsystem on windows if that could cause the issue...

**Comment by pozil** [comment]
Ah yes, I have suspicions on WSL and the Windows file system. It could be that there's an issue with the ways paths are represented vs Unix systems. I'll try to see if I can get my hands on a Windows machine to repro.

**Comment by markocir** [comment]
I tried to run it in Windows cmd but still the same issue persists.
![image](https://user-images.githubusercontent.com/18015314/201894507-57aa4df5-0d27-4f16-88e5-ad006e93f65c.png)

**Comment by markocir** [comment]
Reporting how I managed to get the test running:
1. I replaced import with require 
![image](https://user-images.githubusercontent.com/18015314/201905065-9eaee8d6-94ba-48df-bb4c-e4857dd52086.png)
2. I replaced some paths inside pageObjects to show to correct location; for example: ProductTileList was importing Paginator with path: ebikes-lwc/pageObjects/paginator which was not found changed to (see image)
![image](https://user-images.githubusercontent.com/18015314/201905263-03ccd319-a8d2-4da5-91c1-7e4c5c9268c3.png)

**Comment by pozil** [comment]
Sorry for the delay, I didn't get the chance to run the tests on a Windows machine. I'm glad that you could make it work but fixing code under `pageObjects` is not future proof because it's auto-generated by the UTAM compiler.

There's definitely an issue with path resolution when working with Windows.
Out of curiosity, are the [official JS UTAM recipes](https://github.com/salesforce/utam-js-recipes) working for you?

**Comment by markocir** [comment]
No problem :) 

They are "working": i can run the test but test fail due to some error: unknown error: unknown error: net::ERR_NAME_NOT_RESOLVED -> page crashes.

And only in windows command line. On WSL it is not working due to chromedriver

**Comment by olivier-martin-sf** [comment]
Just adding my 2 cents on the issue:

The error  `Cannot use import statement outside a module` is a `SyntaxError`. Indeed, the use of the `import` keyword (which is part of the JS specification) is only allowed in the context of an ES Module. Using the `import` keyword outside of that context is syntactically invalid. That error is actually thrown by node.js itself when trying to load a file.

Node.js uses some heuristics to [determine the module system](https://nodejs.org/dist/latest-v18.x/docs/api/packages.html#determining-module-system) of a given file. There are a couple of things at play here:

First, we have to look at the root `package.json` to see what the value of the `type` property hold. If it's setup to `commonjs` (or absent), node will use it's legacy CommonJS module format by default for the files that end up with the `.js` file extension. That's the case for this project as we don't have a `type` property in the package manifest. 

Our tests, which have a `.js` file extension, will be interpreted by node.js as being authored with the CommonJS module format. 

> Note: The commonJS module format isn't part of the JS specification and is using a `require` function that's actually backed into node.js. 

So the error is actually what would node.js throw if we were to execute the file directly through it. However, there's a trick in that repo in the WDIO config. We register a `helpers` script that will use Babel to transpile the content of the test file from ES Modules to CommonJS on the fly while running the test. So even if the test uses import it's actually being transpiled into a require call when passed to the node.js runtime.

My guess is that this helper doesn't resolve correctly on your system. Could you please provide the output of: 
`console.log(path.resolve(process.cwd(), 'wdioJasmineHelper.js'));` 

Just to double check if that's the issue you were having. I understood you solve the issue switching to `require` calls but you shouldn't have to.

**Comment by markocir** [comment]
I tried to put the console log into wdioJasmineHelper._jasmineRegister() but nothing was printed out so I put it inside the tests and this is what I got, I think its ok?
![image](https://user-images.githubusercontent.com/18015314/202001724-d736af07-28ff-489f-b462-96cc8b3cc0d9.png)

I can successfully run tests with `yarn test --spec force-app/test/record-create.spec.js` (https://github.com/salesforce/utam-js-recipes)
.

**Comment by olivier-martin-sf** [comment]
Yes that looks correct to me. 

Can you try `npm run test:ui --spec force-app/test/utam/page-explorer.spec.js`?
I would also just try to add a dummy `console.log('something')` statement in the `wdioJasmineHelper.js` file (in the top-level scope) and see if that prints when you start running the test.

**Comment by muenzpraeger** [comment]
Just adding that I see in the OP that `@babel/register` is not loaded.

Instead of `@wdio/config:utils: Found '@babel/register' package, auto-compiling files with Babel` it's showing ` @wdio/config:utils: Found 'ts-node' package, auto-compiling TypeScript files`.

**Comment by muenzpraeger** [comment]
I did a quick investigation in the `wdio` source code. The `ts-node` message is very specific (see [here](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-config/src/utils.ts#L142)).

After a quick test I can reproduce the error when adding `ts-node` to the project (which confirms what @olivier-martin-sf mentioned).

**Comment by markocir** [comment]
Sorry for late reply.

@olivier-martin-sf No I have not changed anything.

This is the package.json:
 ```json
{
    "name": "ebikes-lwc",
    "private": true,
    "version": "1.0.0",
    "description": "EBikes Sample App - Lightning Web Components",
    "scripts": {
        "lint": "eslint **/lwc/**/*.js",
        "test": "npm run test:unit",
        "test:unit": "sfdx-lwc-jest --skipApiVersionCheck",
        "test:unit:watch": "sfdx-lwc-jest --watch --skipApiVersionCheck",
        "test:unit:debug": "sfdx-lwc-jest --debug --skipApiVersionCheck",
        "test:unit:coverage": "sfdx-lwc-jest --coverage --skipApiVersionCheck",
        "test:ui": "wdio",
        "test:ui:compile": "utam -c utam.config.js",
        "test:ui:generate:login": "node scripts/generate-login-url.js",
        "prettier": "prettier --write \"**/*.{cls,cmp,component,css,html,js,json,md,page,trigger,xml,yaml,yml}\"",
        "prettier:verify": "prettier --check \"**/*.{cls,cmp,component,css,html,js,json,md,page,trigger,xml,yaml,yml}\"",
        "postinstall": "husky install",
        "precommit": "lint-staged"
    },
    "lint-staged": {
        "**/*.{cls,cmp,component,css,html,js,json,md,page,trigger,xml,yaml,yml}": [
            "prettier --write"
        ],
        "**/lwc/**/*.js": [
            "eslint"
        ],
        "**/lwc/**": [
            "sfdx-lwc-jest --skipApiVersionCheck -- --bail --findRelatedTests --passWithNoTests"
        ]
    },
    "author": "salesforce.com",
    "license": "CC0-1.0",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/trailheadapps/ebikes-lwc"
    },
    "devDependencies": {
        "@babel/cli": "^7.19.3",
        "@babel/core": "^7.20.2",
        "@babel/preset-env": "^7.20.2",
        "@babel/register": "^7.18.9",
        "@lwc/eslint-plugin-lwc": "^1.5.0",
        "@prettier/plugin-xml": "^2.2.0",
        "@sa11y/jest": "^4.0.1",
        "@salesforce/eslint-config-lwc": "^3.3.3",
        "@salesforce/eslint-plugin-lightning": "^1.0.0",
        "@salesforce/sfdx-lwc-jest": "^1.1.3",
        "@wdio/cli": "^7.25.4",
        "@wdio/jasmine-framework": "^7.25.4",
        "@wdio/local-runner": "^7.25.4",
        "@wdio/spec-reporter": "^7.25.4",
        "chromedriver": "^107.0.3",
        "dotenv": "^16.0.3",
        "eslint": "^8.27.0",
        "eslint-plugin-import": "^2.26.0",
        "eslint-plugin-jest": "^27.1.4",
        "husky": "^8.0.1",
        "lint-staged": "^13.0.3",
        "prettier": "^2.7.1",
        "prettier-plugin-apex": "^1.11.0",
        "salesforce-pageobjects": "^2.2.0",
        "utam": "^1.2.1",
        "wdio-chromedriver-service": "^8.0.0",
        "wdio-utam-service": "^1.2.1"
    },
    "workspaces": [
        "./"
    ],
    "volta": {
        "node": "16.13.2",
        "npm": "8.1.2"
    }
}

 ```

**Comment by markocir** [comment]
How can I test from where ts-node is comming from? Nor bash or cmd recognize ts-node or typescript command.

**Comment by olivier-martin-sf** [comment]
`npm explain ts-node` should give you some information about this!

**Comment by markocir** [comment]
![image](https://user-images.githubusercontent.com/18015314/204089274-1121279e-12a6-41db-aaf1-2873fb988d1d.png)

**Comment by olivier-martin-sf** [comment]
I would try 2 things:

1. First delete your `node_modules` folder and try to run `npm install && npm test:ui:compile`
2. Disable WDIO auto compile option by adding ` autoCompileOpts: { autoCompile: false }` in the `config` object that's exported by `wdio.conf.js`

**Comment by markocir** [comment]
Thank you everybody! 

I discovered that I had `node_module` in `a` folder of folder structure: `a/b/c/ebikes`

When running `npm explain ts-node` in `ebikes` folder only `ebikes/node_modules` folder is evaluated which in fact did not contain `ts-node` module. When running `npm explain ts-node` in `c` folder `ts-node` was detected. 

So I just removed the `a/node_modules` folder and reinstalled the `ebikes` and now it works.

Thanks again for your help!
